api/v1alpha1/common.go

```diff
@@ -251,3 +253,49 @@ const (
 	SourceComponent   string = "source"
 	SinkComponent     string = "sink"
 )
+
+// Config represents untyped YAML configuration.
+type Config struct {
+	// Data holds the configuration keys and values.
+	// This field exists to work around https://github.com/kubernetes-sigs/kubebuilder/issues/528
+	Data map[string]interface{} `json:"-"`
+}
+
+// NewConfig constructs a Config with the given unstructured configuration data.
+func NewConfig(cfg map[string]interface{}) Config {
+	return Config{Data: cfg}
+}
+
+// MarshalJSON implements the Marshaler interface.
+func (c *Config) MarshalJSON() ([]byte, error) {
+	return json.Marshal(c.Data)
+}
+
+// UnmarshalJSON implements the Unmarshaler interface.
+func (c *Config) UnmarshalJSON(data []byte) error {
+	var out map[string]interface{}
+	err := json.Unmarshal(data, &out)
+	if err != nil {
+		return err
+	}
+	c.Data = out
+	return nil
+}
+
+// DeepCopyInto is an ~autogenerated~ deepcopy function, copying the receiver, writing into out. in must be non-nil.
+// This exists here to work around https://github.com/kubernetes/code-generator/issues/50
+func (c *Config) DeepCopyInto(out *Config) {
+	bytes, err := json.Marshal(c.Data)
```

Comment:
Could https://github.com/kubernetes/apimachinery/blob/master/pkg/runtime/converter.go#L437 be more efficient?